contracts/ERC1949.sol

```diff
@@ -17,13 +17,17 @@ contract ERC1949 is IERC1949, ERC1948 {
     data[queenId] = bytes32(uint256(1));
   }
 
-  modifier onlyQueenOwner() {
-    require(queenOwners[msg.sender] > 0, "sender not queen owner");
+  modifier onlyQueenOwner(address _to) {
+    require(
+        (queenOwners[msg.sender] > 0) ||
+        ((queenOwners[_to] > 0) && _isApprovedOrOwner(msg.sender, queenOwners[_to])),
+        "sender not queen owner nor approved"
+    );
     _;
   }
 
-  function breed(uint256 _workerId, bytes32 _workerData) public onlyQueenOwner() {
-    super._mint(msg.sender, _workerId);
+  function breed(uint256 _workerId, address _to, bytes32 _workerData) public onlyQueenOwner(_to) {
```

Comment:
@johannbarbie
I updated this to support exit and the `allowance` stuff.
Personally, I did like it to be `breed(address,uint256,bytes32)` but I left it this way because the leap-node contract already used this signature 🤷‍♂️

Reply:
valid way to do it, but you should also adjust it here: github.com/ethereum/EIPs/pull/1949